src/IR/types.ts

```diff
@@ -54,3 +57,27 @@ export function arrayType(
     length,
   };
 }
+
+export function integerType(
+  low?: bigint | number,
+  high?: bigint | number
+): ValueType {
+  return {
+    type: "integer",
+    low: typeof low === "number" ? BigInt(low) : low,
+    high: typeof high === "number" ? BigInt(high) : high,
+  };
+}
+
+export function integerTypeFromMultipleBounds(bounds: bigint[]) {
```

Comment:
This might be better named as a union of bounds since we might later need an intersection of bounds